sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -89,10 +89,12 @@ class V2SessionCatalog(catalog: SessionCatalog)
           throw QueryCompilationErrors.timeTravelUnsupportedError(
```

Comment:
do we still need the `if-else`?

Reply:
Yeah, after merging, the code here can be simplified. It's my bad, let me do it immediately.

Reply:
@cloud-fan 
Follow Up pr: https://github.com/apache/spark/pull/42082